woocommerce/google-listings-and-ads
# Use WooCommerce CRUD class methods to handle metadata

### Changes proposed in this Pull Request:



Closes #694.

This PR modifies the usages of WordPress metadata functions in favor of using WooCommerce's CRUD methods. It also contains some extra changes to the GlobalSiteTag class to inject the services it uses directly and remove the dependency on the container class.

### Detailed test instructions:

The following functionality must be working correctly:
1. Sync a product and check the sync-related metadata (`_wc_gla_sync_status`, `_wc_gla_google_ids`, etc.) exists in database
2. Un-sync a product and make sure the said metadata gets deleted from the database
3. Update a product's attributes and make sure they are updated in DB
4. Delete a product's attributes (set it to empty) and make sure it's reflected in DB
5. Make sure the `gtag` script to send the conversion event is included in the order confirmation page (related to the change in `GlobalSiteTag`)
